### PR TITLE
[Android] Splash screen in standalone app fixes

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Resolves expo/expo#2533
Corresponding docs for https://github.com/expo/expo-cli/pull/268

- [ ] update `xdl` package in `tool-public` to one containing above `expo-cli` changes

### CODE REVIEW

File: docs/versions/unversioned/sdk/splash-screen.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ export default class App extends React.Component {
 }
 ```
 
-## Example without any flickering between SplashScreen and it's later continueation
+## Example without any flickering between SplashScreen and it's later continuation
```
Comment (bbarthec) [inline docs/versions/unversioned/sdk/splash-screen.md]: This is only explicit change I've made, all the rest is just auto-behaviour of my IDE

File: docs/versions/unversioned/workflow/configuration.md [hunk 2]
```diff
@@ -670,23 +670,26 @@ Configuration for how and when the app should request OTA JavaScript updates
       "backgroundColor": STRING,
 
       /*
-        Determines how the "image" will be displayed in the splash loading screen. 
-        Must be one of "cover" or "contain", defaults to "contain". 
+        Determines how the "image" will be displayed in the splash loading screen.
```
Comment (bbarthec) [inline docs/versions/unversioned/workflow/configuration.md]: My explicit changes are here, the rest is just out-fixing made by my IDE

File: docs/versions/unversioned/guides/splash-screens.md [hunk 3]
```diff
@@ -89,6 +89,18 @@ Your app can be opened from the Expo client or in a standalone app, and it can b
...
 
+### Differences between environments - Android
+
+Splash screen behaves in most cases exactly the same as in iOS case.
+
+There is slightly difference when it comes down to **standalone Android applications**.
+In this scenario extra attention should be paid to [Android.`splash` section](../workflow/configuration.html#android) configuration provided by [`app.json`](../workflow/configuration.html).
+
+Depending on the `resizeMode` you choose you will get followings:
+- **cover** - In this mode Expo is leveraging Android possibility to present static bitmap at the very beginning of the application start. Unfortunately Android (unlike iOS) is not supporting provided image streching, so Expo will just present given image centered on the screen.
```
Comment (tsapeta) [inline docs/versions/unversioned/guides/splash-screens.md]:
```suggestion
- **cover** - In this mode Expo is leveraging Android possibility to present static bitmap at the very beginning of the application start. Unfortunately Android (unlike iOS) is not supporting provided image stretching, so Expo will just present given image centered on the screen.
```

File: docs/versions/unversioned/guides/splash-screens.md [hunk 4]
```diff
@@ -89,6 +89,18 @@ Your app can be opened from the Expo client or in a standalone app, and it can b
...
+By default `splash.image` would be used as `mdpi` resource. It's up to you to provide graphics that meet your expectations and fits the screen dimension. To achieve this use different resolutions for [different device DPIs](configuration.html#mdpi), from `mdpi` to `xxxhdpi`.
```
Comment (tsapeta) [inline docs/versions/unversioned/guides/splash-screens.md]:
```suggestion
By default `splash.image` would be used as `mdpi` resource. It's up to you to provide graphics that meet your expectations and fit the screen dimension. To achieve this, use different resolutions for [different device DPIs](configuration.html#mdpi), from `mdpi` to `xxxhdpi`.
```

File: docs/versions/unversioned/guides/splash-screens.md [hunk 5]
```diff
@@ -89,6 +89,18 @@ Your app can be opened from the Expo client or in a standalone app, and it can b
...
+- **contain** - As described in _cover_ mode we're unable to dynamicly adjust image to screen size at the very beginning of the application start. Therefore in this mode only background color will be presented and a while later when some view hierarchy is mounted we'll show provided `splash.image`.
```
Comment (tsapeta) [inline docs/versions/unversioned/guides/splash-screens.md]:
```suggestion
- **contain** - As described in _cover_ mode we're unable to dynamically adjust image to screen size at the very beginning of the application start. Therefore in this mode only background color will be presented and a while later when some view hierarchy is mounted we'll show provided `splash.image`.
```

**Review by tsapeta — CHANGES_REQUESTED** [review]
Added suggestions about some typos 😉

**Review by esamelson — COMMENTED** [review]
Looks good, can you make sure to add a CHANGELOG entry as well since this may be a breaking change for people (they may need to switch from "contain" to "cover" or vice versa)?

**Review by sjchmiela — COMMENTED** [review]
Nice one, I don't know if all my suggestions make sense. Nice job! 💯

**Pull request merged**